tests/testthat/_snaps/logging.md

```diff
@@ -198,3 +198,148 @@
+      A: x5   B: x5
+
+# interactive logger works (fit_resamples, rlang warning + error)
+
+    Code
+      res_fit <- fit_resamples(parsnip::nearest_neighbor("regression", "kknn"),
+      Sale_Price ~ ., rsample::vfold_cv(modeldata::ames[, c(72, 40:45)], 5), control = control_resamples(
+        extract = function(x) {
+          raise_warning_rl()
+          raise_error_rl()
+        }))
+    Message
+      > A | warning: ope! yikes. (but rlang)
+      > B | error:   AHHhH (but rlang)
+
+---
+
+    Code
+      catalog_summary_test
+    Output
+      A: x5   B: x5
+
+# interactive logger works (fit_resamples, multiline)
+
+    Code
+      res_fit <- fit_resamples(parsnip::nearest_neighbor("regression", "kknn"),
+      Sale_Price ~ ., rsample::vfold_cv(modeldata::ames[, c(72, 40:45)], 5), control = control_resamples(
+        extract = raise_multiline_conditions))
+    Message
+      > A | warning: hmmm what's happening
+      > B | error:   aHHHksdjvndiuf
+
+---
+
+    Code
+      catalog_summary_test
+    Output
+      A: x5   B: x5
+
+# interactive logger works (fit_resamples, occasional error)
+
+    Code
+      res_fit <- fit_resamples(parsnip::nearest_neighbor("regression", "kknn"),
+      Sale_Price ~ ., rsample::vfold_cv(modeldata::ames[, c(72, 40:45)], 5), control = control_resamples(
+        extract = later))
+    Message
+      > A | error:   this errors now! ha!
+
+---
+
+    Code
+      catalog_summary_test
+    Output
+      A: x2
+
+# interactive logger works (fit_resamples, occasional error + warning)
+
+    Code
+      res_fit <- fit_resamples(parsnip::nearest_neighbor("regression", "kknn"),
+      Sale_Price ~ ., rsample::vfold_cv(modeldata::ames[, c(72, 40:45)], 10),
+      control = control_resamples(extract = function(x) {
+        once()
+        later()
+      }))
+    Message
+      > A | error:   oh no
+      > B | error:   this errors now! ha!
+
+---
+
+    Code
+      catalog_summary_test
+    Output
+      A: x1   B: x6
+
+# interactive logger works (fit_resamples, many distinct errors)
+
+    Code
+      res_fit <- fit_resamples(parsnip::nearest_neighbor("regression", "kknn"),
+      Sale_Price ~ ., rsample::vfold_cv(modeldata::ames[, c(72, 40:45)], 5), control = control_resamples(
+        extract = numbered))
+    Message
+      > A | error:   error number 1
+      > B | error:   error number 2
+      > C | error:   error number 3
+      > D | error:   error number 4
+      > E | error:   error number 5
+
+---
+
+    Code
+      catalog_summary_test
+    Output
+      A: x1   B: x1   C: x1   D: x1   E: x1
+
+# interactive logger works (tune grid, error)
+
+    Code
+      res_fit <- tune_grid(parsnip::nearest_neighbor("regression", "kknn", neighbors = tune()),
+      Sale_Price ~ ., rsample::vfold_cv(modeldata::ames[, c(72, 40:45)], 5), grid = 5,
+      control = control_grid(extract = raise_error))
+    Message
+      > A | error:   AHHhH
+
+---
+
+    Code
+      catalog_summary_test
+    Output
+      A: x5
```

Comment:
Poked at this for a moment and suspicious this might be a submodel trick thing... confirming now, if so will add a comment to assuage future wonk

Reply:
Yup, submodel wonk🤪 Extraction happens once per model fit, only one model fit per resample. I changed the argument to tune in 8b128a02 so that that count is more self-evident

Reply:
Ooooohhh! Okay, awesome! Excellent change.